### PR TITLE
Use the new deleteEntityRecord to delete menus

### LINKED ISSUE #22340: Deleted menus reappear when switching to Menu Locations tab and back again
**Describe the bug**
On the experimental navigation menu page, deleted menus reappear when switching to Menu Locations tab and back again.

**To reproduce**
1. Add some menus
2. Delete one
3. Select 'Menu Locations'
4. Go back to 'Edit Navigation'
5. Note that the deleted menu is back in the Select Navigation menu.

**Expected behavior**
The deleted menu doesn't reappear

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Closes #22340 This PR implements deleting menus using entity delete. It depends on #21557

## How has this been tested?
Tested locally by:

1. Creating menus
2. Deleting menus

### CODE REVIEW

File: packages/core-data/src/queried-data/reducer.js [hunk 1]
```diff
@@ -89,6 +89,19 @@ function items( state = {}, action ) {
 	return state;
 }
 
+const removeQueryItems = () => ( reducer ) => ( state, action ) => {
+	const reconciledState = { ...state };
+	if ( action.type === 'REMOVE_ITEMS' ) {
+		action.query.forEach( ( queryId ) => {
+			remove( reconciledState[ '' ], ( stateQueryId ) => {
```
Comment (epiqueras) [inline packages/core-data/src/queried-data/reducer.js:96]: The state at this point contains all the queries, keyed by `stableKey`. Here you are just working with the empty query. This logic needs to work for every query and should probably be in the sub-keyed reducer.

**Review by epiqueras — CHANGES_REQUESTED** [review]

File: packages/core-data/src/queried-data/reducer.js (hunk 1 above)
Comment (draganescu) [reply packages/core-data/src/queried-data/reducer.js:96]: I [moved this logic](https://github.com/WordPress/gutenberg/pull/21557/files#diff-f7aec7a322c8fd5825dcb818fc148a9cR33) into the `onSubKey` reducer, got a bit more complex. Will take it out of this PR.

Comment (epiqueras) [reply packages/core-data/src/queried-data/reducer.js]: `onSubKey` is a generic utility. This doesn't belong there.

>should probably be in the sub-keyed reducer.

That's the reducer that gets passed to the function returned by `flowRight`.

**Comment by draganescu** [comment]
@adamziel 's #22603 makes this useless. Not sure whether to close it or not.

**Comment by adamziel** [comment]
@draganescu not really, #22603 is all about menu items and this one is about menus. Using batch requests for menu items makes sense since the model of interaction with that screen is "make a bunch of changes and save them all at once. On the other hand, the model of interaction with menus is more like "change one thing and then save it immediately" so this PR is not only useful, but would also solve #22340.

**Review by adamziel — APPROVED** [review]

**Pull request merged**